src/components/Table/index.tsx

```diff
@@ -155,9 +156,14 @@ class Table<T> extends React.Component<TableProps<T>, TableState<T>> {
 		};
 	}
 
-	componentWillReceiveProps(newProps: TableProps<T>) {
-		if (newProps.sort !== this.props.sort) {
-			this.setState({ sort: newProps.sort } as TableState<T>);
+	static getDerivedStateFromProps<T>(
```

Comment:
I tried this out and it seems we cannot use `getDerivedStateFromProps` properly, because it is also called when you call `setState` and we cannot know which one we should set as the sort value (there isn't a single source of truth). After all, it seems the best approach for now is to use 
```
	componentDidUpdate(prevProps: TableProps<T>) {
		if(this.props.sort && prevProps.sort !== this.props.sort){
			this.setState({sort: this.props.sort});
		}
	}
```
instead, I shouldn't have insisted on using `getDerivedStateFromProps` without trying it out properly.

Reply:
As I said from the beginning =)